pkg/aws/instance_info_test.go

```diff
@@ -0,0 +1,41 @@
+package aws
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestInstanceInfo(t *testing.T) {
+	for _, tc := range []Instance{
+		{
+			InstanceType: "m4.xlarge",
+			VCPU:         4,
+			Memory:       17179869184,
+		},
+		{
+			InstanceType:           "i3.4xlarge",
+			VCPU:                   16,
+			Memory:                 130996502528,
+			NVMEInstanceStorage:    true,
+			InstanceStorageDevices: []StorageDevice{{Path: "/dev/nvme0n1", NVME: true}, {Path: "/dev/nvme1n1", NVME: true}},
+		},
+		{
+			InstanceType:           "m5d.4xlarge",
+			VCPU:                   16,
+			Memory:                 68719476736,
+			NVMEInstanceStorage:    false,
+			InstanceStorageDevices: []StorageDevice{{Path: "/dev/nvme1n1", NVME: true}, {Path: "/dev/nvme2n1", NVME: true}},
```

Comment:
Another test with `instance.Storage.NeedsInitialization=True` would be nice to cover all cases